aws_lambda_powertools/metrics/provider/cloudwatch_emf/cloudwatch.py

```diff
@@ -272,10 +272,18 @@ def add_dimension(self, name: str, value: str) -> None:
             raise SchemaValidationError(
                 f"Maximum number of dimensions exceeded ({MAX_DIMENSIONS}): Unable to add dimension {name}.",
             )
-        # Cast value to str according to EMF spec
-        # Majority of values are expected to be string already, so
-        # checking before casting improves performance in most cases
-        self.dimension_set[name] = value if isinstance(value, str) else str(value)
+
+        if not name or not value:
```

Comment:
Correct me if I'm wrong - this check would pass if, for some reason, `value` is a string `" "`?

Reply:
Good catch @dreamorosi! Fixed